src/smtml/mappings.ml

```diff
@@ -66,7 +66,7 @@ module Make (M_with_make : M_with_make) : S_with_fresh = struct
 
     let make_symbol (ctx : symbol_ctx) (s : Symbol.t) : symbol_ctx * M.term =
       let name = match s.name with Simple name -> name | _ -> assert false in
-      if M.caches_consts then
+      if not M.caches_consts then
```

Comment:
```suggestion
      if not M.caches_consts then
```

I think this should remain without the `not` no? If the solver doesn't cache consts we want to do the lookup in the else branch to ensure we're not creating fresh symbols when the have the same name

Reply:
Right I assumed that it was the mapping that was meant to cache the consts, I will fix it